app/models/borrow_direct_reshare_requests.rb

```diff
@@ -0,0 +1,118 @@
+# frozen_string_literal: true
+
+class BorrowDirectReshareRequests
+  attr_reader :patron
+
+  def initialize(patron)
+    @patron = patron
+  end
+
+  def requests
+    # Ensure ReShare requests are active (not yet in Symphony)
+    # and that each ReShare request object's patron_id matches
+    # the logged in patron's university_id.
+    reshare_requests.select(&:active?)
+                    .select { |request| request.patron_id == patron.university_id }
+  end
+
+  private
+
+  def reshare_requests
+    (request_client.requests(patron.university_id) || []).map do |request|
```

Comment:
How about having `requests()` return an array (rather than nil)?

Reply:
I like it.